src/compiler/binder.ts

```diff
@@ -2568,6 +2578,18 @@ namespace ts {
             }
         }
 
+        function bindDynamicallyNamedThisPropertyAssignment(node: BinaryExpression | DynamicNamedDeclaration, symbol: Symbol) {
+            bindAnonymousDeclaration(node, SymbolFlags.Property, InternalSymbolName.Computed);
+            addLateBoundAssignmentDeclarationToSymbol(node, symbol);
+        }
+
+        function addLateBoundAssignmentDeclarationToSymbol(node: BinaryExpression | DynamicNamedDeclaration, symbol: Symbol | undefined) {
+            if (symbol) {
+                const members = symbol.assignmentDeclarationMembers || (symbol.assignmentDeclarationMembers = createMap());

```

Comment:
changing the shape of `Symbol` makes me think we should run the perf tests as a precaution.